### ISSUE TITLE
number-unit-format treats a decimal fraction as a unit, so any bare decimal ("5.6", "3.11") is a false positive

### ISSUE THREAD

**Issue by Jamie-BitFlight** [Jamie-BitFlight/textlint-rule-preset-ste-ai#141]
## Symptom

`ste-ai/number-unit-format` reads the fractional part of a decimal number as if it were a unit, and
demands a space before it. Writing a version number, or any bare decimal with no unit after it,
produces a finding that cannot be satisfied without corrupting the number.

```
The GPT-5.6 series is here.
  → Put a space between the number and the unit: "5 .6".

Use version 5.6 of the tool.
  → Put a space between the number and the unit: "5 .6".

Install Python 3.11 first.
  → Put a space between the number and the unit: "3 .11".
```

The suggested repair is "5 .6", which is not a number.

## What is not affected

The rule behaves correctly when a real unit follows, and when the number is an integer. Both of
these are clean:

```
The file is 5.6 MB in size.     → no finding
Node 22 is supported.           → no finding
```

So the trigger is specifically a decimal whose fractional part is not followed by a unit.

## Reproduction

```bash
npx -p vite-plus@0.3.0 vp install --frozen-lockfile
node_modules/.bin/vp pack
printf 'Install Python 3.11 first.\n' > probe.md
node_modules/.bin/textlint --config .textlintrc.json -f compact probe.md
```

Observed on head `7d9a438`:

```
probe.md: line 1, col 16, Error - [deterministic-violation][provisional] Put a space between the
number and the unit: "3 .11". (ste-ai/number-unit-format)
```

## How it was found

While adding a sentence to `AGENTS.md` that named a vendor's model series by version. The sentence
was rejected by the repo's own dogfood ratchet with `[ste-ai/number-unit-format]: 0 -> 1`. The
version reference was dropped to get the change through, which is a workaround rather than a fix.

## Where

`src/deterministic/rules/mechanics.ts`'s `numberUnitFormatRule`. Its `run` iterates
`doc.protectedRegions` for regions of kind `numeric-expression`, and this rule is the one that
deliberately reads inside a protected numeric expression — its own `numberUnitMeta.description`
says so: "This is the one rule that reads inside protected numeric expressions." The decimal point
appears to be splitting the numeric expression into a quantity and a trailing token that is then
treated as a unit candidate. That last clause is a hypothesis about the mechanism, not a verified
claim; the observed behaviour above is what is verified. To confirm, instrument the unit-candidate
extraction and check what token it yields for `3.11`.

## Impact

Two things make this worse than a cosmetic nit:

1. **It is unsatisfiable.** Every other finding this preset raises can be fixed by rewriting. This
   one cannot, short of deleting the number or `.textlintignore`-ing the file, and neither is a
   legitimate use of that ignore list per its own header comment.
2. **It surfaces at `Error` severity**, so it blocks the CI ratchet in
   `scripts/ci/check-dogfood-lint.mjs`. Note that `numberUnitMeta` declares
   `defaultSeverity: 'warning'` — the escalation to `Error` looks like the behaviour already
   tracked in #136 ("Apply RuleMetadata.defaultSeverity to runtime diagnostics"). If #136 is fixed
   first, this drops from blocking to advisory, but the false positive itself remains.

Decimal version numbers are common in this repository's own documentation, so this will keep
recurring.

## Related

- #136 — `defaultSeverity` is not applied to runtime diagnostics, which is why this warning-level
  rule blocks CI as an error.
- Not a duplicate of any open issue I could find; I listed all 39 open issues and none names this
  rule.

**Comment by Jamie-BitFlight** [comment]
## Grooming notes

**Root cause confirmed.** `numberUnitFormatRule` in `src/deterministic/rules/mechanics.ts` runs this regex against each `numeric-expression` region: `/^([+±-]?\d+(?:[.,]\d+)?)(\s*)([^\s\d].*)$/`. Group 3, the unit, is mandatory. Running it directly against the reported inputs gives:

```
"3.11"   -> quantity: "3",   gap: "",  unit: ".11"
"5.6"    -> quantity: "5",   gap: "",  unit: ".6"
"5.6 MB" -> quantity: "5.6", gap: " ", unit: "MB"
"22"     -> no match
```

For a bare decimal, group 1's optional `(?:[.,]\d+)?` clause first grabs the whole number. That leaves nothing for the mandatory group 3, so the engine backtracks: group 1 gives back the fractional part, and the leading `.` then satisfies `[^\s\d]` as a false unit. This confirms the issue's own hypothesis as fact.

The `numeric-expression` protected-region pass's second pattern in `src/core/protected-regions.ts` is the one that captures a bare version-style decimal like `3.11` as its own region with nothing trailing, so `raw` really is just the number — there is no real unit for the rule to find.

**Coverage gap.** `test/unit/rules.test.ts`'s `number-unit-format` suite has no case for a bare decimal with no trailing unit. Every existing case has either a real unit (`25Nm`, `25 Nm`, `0,5 mm`) or a percent sign. That gap is why this shipped unnoticed.

**Suggested fix shape.** Make the unit portion of the regex optional as a whole group, not just its internal digits, for example `^([+±-]?\d+(?:[.,]\d+)?)(?:(\s*)([^\s\d].*))?$`. When group 3 is `undefined`, there is no unit to check, so the rule should skip the spacing diagnostic rather than relying on the whole match failing. The decimal-comma check further down is a separate regex and is not affected by this bug.

**Classification:** defect in a deterministic, provisional-status rule. Reproducible, root cause located precisely, fix shape identified. This is plannable without further research.

**Relation to #136:** the cross-reference in the issue body is correct. The two problems are independent. Fixing #136 first only demotes this from CI-blocking to advisory — the false positive itself stays until this issue is fixed on its own.

**Suggested acceptance criteria:**
- `Install Python 3.11 first.` and `Use version 5.6 of the tool.` produce no `number-unit-format` finding.
- `The file is 5.6 MB in size.` still produces no finding (must not regress).
- `Torque the bolt to 25Nm now.` still flags the missing space (must not regress).
- A regression test covers a bare multi-segment version number too (for example `1.2.3`), since the same protected-region pattern captures those the same way.

---
_Generated by [Claude Code](https://claude.ai/code)_
